### ISSUE TITLE
Clarify in docs how and where prepick is applied to detections

### ISSUE THREAD

**Issue by RensHofman** [eqcorrscan/EQcorrscan#255]
Description:
The templates prepick time is not added to the detection time.

To reproduce:
I create templates from P-picks using template_gen.from_meta_file. In this function, I set the prepick attribute (to 0.15) so that my template starts just before the pick. After saving the templates to files, I load them into match_filter.Template objects, thereby setting the prepick attribute to the same value as before. When using the template to detect matches, the detection time points to the start of the template waveform, not taking into account the prepick time. 

This becomes most evident when self-detecting the template in its original trace. The detection time will be the original P-pick time from which the template was cut minus the prepick time.

Expected behaviour:
I would expect the prepick time to be added to the detection time, so that it correlates with the original pick on which the template is based, rather than the starttime of the template.

**Comment by calum-chamberlain** [comment]
Hi @RensHofman, thanks for putting this here.  However, this is the expected behaviour.  If you want to change the detection time yourself afterwards then you should.

If we correct for pre-pick then the detection-time corresponds to the first pick in the template.  Consider a case whereby the first picked station is not available, the corrected detection time would then correspond to something that isn't there. The more useful thing for earthquakes would be to report origin-time, but for many reasons (not least that we don't always have an origin) we don't do that.

We could correct the picks in the `detection.event` attribute, but don't, for backwards compatibility (see warning [here](http://eqcorrscan.readthedocs.io/en/latest/submodules/core.match_filter.Tribe.html#eqcorrscan.core.match_filter.Tribe.detect).

To summarise, the detection-time is what I think it should be (the time of the detection), this is not corrected for pre-pick, and I don't think it should be. However, this could/should be made clearer in the docs.

**Comment by calum-chamberlain** [comment]
@RensHofman, I was thinking of making a note on the `Tribe.detect`, `Tribe.client_detect` and `match_filter` pages saying something along the lines of:

> Note that the reported detection-time corresponds to the time of the beginning of the earliest template waveform at detection, not the pick-time (the pre-pick is not corrected for).

Do you have any suggestions for better wording or where you would like to see this in the docs?

**Comment by RensHofman** [comment]
Hi @calum-chamberlain, thanks for your reply.  I understand your point. The confusing part for me was that I parse my locally stored templates into a Tribe object, and that I have the option to set the pre-pick attribute at this point. From this I inferred that the attribute has some use which it doesn't seem to have.

The note clearly explains it for me.

**Comment by calum-chamberlain** [comment]
That is fair - in the detection stage the pre-pick is mostly used for book-keeping (so that templates can be re-created if needed), however, the pre-pick is used in [`Party.lagcalc`](http://eqcorrscan.readthedocs.io/en/latest/submodules/core.match_filter.Party.html#eqcorrscan.core.match_filter.Party.lag_calc).

Is there a reason that you chose not to use the [`Tribe.construct`](http://eqcorrscan.readthedocs.io/en/latest/submodules/core.match_filter.Tribe.html#eqcorrscan.core.match_filter.Tribe.construct) method to generate the templates in the first place?
